tools/check_for_upgrade.sh

```diff
@@ -1,13 +1,17 @@
 #!/usr/bin/env zsh
 
+if [ -z "$ZSH_UPDATE" ]; then
+  ZSH_UPDATE="$ZSH/cache/.zsh-update"
```

Comment:
`$ZSH_CACHE_DIR` should be available here but I haven't checked. Use that instead of `$ZSH/cache` if it works.